utils/src/test/java/org/robolectric/util/SchedulerTest.kt

```diff
@@ -0,0 +1,468 @@
+package org.robolectric.util
+
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.Iterables
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import java.util.Random
+import java.util.TreeMap
+import java.util.concurrent.atomic.AtomicLong
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class SchedulerTest {
+  private val scheduler = Scheduler()
+  private val transcript: MutableList<String> = ArrayList()
+  private var startTime: Long = 0
+
+  @Before
+  @Throws(Exception::class)
+  fun setUp() {
+    scheduler.pause()
+    startTime = scheduler.currentTime
+  }
+
+  @Test
+  fun whenIdleStateIsConstantIdle_isPausedReturnsFalse() {
+    scheduler.idleState = Scheduler.IdleState.CONSTANT_IDLE
```

Comment:
What about importing constants of `Scheduler.IdleState` the same as origin Java code? It will make code more clear.